src/uct/base/uct_iface.c

```diff
@@ -548,3 +549,42 @@ ucs_config_field_t uct_iface_config_table[] = {
 
   {NULL}
 };
+
+ucs_status_t uct_base_ep_am_short_iov(uct_ep_h ep, uint8_t id, const uct_iov_t *iov,
+                                      size_t iovcnt)
+{
+    size_t length = 0;
+    unsigned buffer_length;
+    void *buffer;
+    size_t iov_it;
+    void *payload;
+    ucs_status_t status;
+
+    buffer_length = ucs_max(uct_iov_total_length(iov, iovcnt), sizeof(uint64_t));
+    if (buffer_length > UCS_ALLOCA_MAX_SIZE) {
+        buffer = malloc(buffer_length);
+    } else {
+        buffer = ucs_alloca(buffer_length);
+    }
+    memset(buffer, 0, sizeof(uint64_t));
```

Comment:
Yes, it will be overwritten. The zeroing is required to avoid uninitialized value passing into uct_ep_am_short in case of uct_iov_total_length < sizeof(uint64_t).